compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -1210,49 +1211,96 @@ impl<'local, 'interner> Interpreter<'local, 'interner> {
     }
 
     fn evaluate_for(&mut self, for_: HirForStatement) -> IResult<Value> {
-        // i128 can store all values from i8 - u64
-        let get_index = |this: &mut Self, expr| -> IResult<(_, fn(_) -> _)> {
-            match this.evaluate(expr)? {
-                Value::I8(value) => Ok((value as i128, |i| Value::I8(i as i8))),
-                Value::I16(value) => Ok((value as i128, |i| Value::I16(i as i16))),
-                Value::I32(value) => Ok((value as i128, |i| Value::I32(i as i32))),
-                Value::I64(value) => Ok((value as i128, |i| Value::I64(i as i64))),
-                Value::U8(value) => Ok((value as i128, |i| Value::U8(i as u8))),
-                Value::U16(value) => Ok((value as i128, |i| Value::U16(i as u16))),
-                Value::U32(value) => Ok((value as i128, |i| Value::U32(i as u32))),
-                Value::U64(value) => Ok((value as i128, |i| Value::U64(i as u64))),
-                value => {
-                    let location = this.elaborator.interner.expr_location(&expr);
-                    let typ = value.get_type().into_owned();
-                    Err(InterpreterError::NonIntegerUsedInLoop { typ, location })
-                }
+        let start_value = self.evaluate(for_.start_range)?;
+        let signed = match start_value {
+            Value::I8(_) | Value::I16(_) | Value::I32(_) | Value::I64(_) => true,
+            Value::U1(_)
+            | Value::U8(_)
+            | Value::U16(_)
+            | Value::U32(_)
+            | Value::U64(_)
+            | Value::U128(_) => false,
```

Comment:
Can you move out this explicit match for signedness? Either to a helper function or we can do:
```rs
let index_type = start_value.get_type();
if index_type.is_signed() {

} else if index_type.is_unsigned() {

} else {
    Err(...)
}